README.md

```diff
@@ -1,17 +1,121 @@
-# jujulib
 
-This project provides a JavaScript API client for interacting with the Juju
+# JS Jujulib
+
+This project provides a JavaScript API client library for interacting with the Juju
 WebSocket API.
 
-Take a look at the [documentation](api/doc/index.md) for instructions on how to
-get started with the library.
+- [Getting Started](#getting-started)
+- [Client API Reference](#client-api-reference)
+- [Facade API Reference](#facade-api-reference)
+- [Examples](#examples)
+- [Library Maintenance](#library-maintenance)
+  - [Updating Library Facades](#updating-library-facades)
+  - [Releasing to NPM](#releasing-to-npm)
+
+## Getting Started
+
+In order to access the Juju API, a connection must be made, to either a Juju
+controller or model.
+
+```javascript
+import ApplicationV5 from "jujulib/api/facades/application-v5.js";
+import ClientV1 from "jujulib/api/facades/client-v1.js";
+import ClientV2 from "jujulib/api/facades/client-v2.js";
+
+const juju = await jujulib.connect(
+  'wss://example.com/model/x-y-z/api',
+  {facades: [ApplicationV5, ClientV1, ClientV2]}
+);
+```
+
+In the code above, a connection is being established to the provided model API URL, and the client declares to be interested in using the specified set of facades. When multiple versions of the same facade are included (like the two client versions in the example), the most recent version supported by the server is made available to the client.
+
+The `connect` method returns a `juju` object which can be used to log into the controller or model by providing a userpass credentials or macaroons. See The various [examples](#examples) below.
```

Comment:
Can you link to the "juju object" reference docs

Reply:
Yeah I can do that, see above about needing the 'actual' url. I'll do this as a follow-up.